Create (untested) libascon

Denton-L opened the pull request.

Denton-L committed:
Create (untested) libascon

tianyouw commented on projects/briey/libs/ascon.c:
```diff
@@ -0,0 +1,87 @@
+		;
+}
+
+static void ascon_memcpy(volatile uint32_t *dest, const volatile uint32_t *src, size_t size)
```
Add restrict to the pointer qualifiers since we shouldn't be mem copying overlapping memory.

tianyouw commented on projects/briey/libs/ascon.c:
```diff
@@ -0,0 +1,87 @@
+#include <stdint.h>
+#include <string.h>
+
+static volatile struct {
```
Maybe comment/describe each member.

tianyouw commented on projects/briey/libs/ascon.h:
```diff
@@ -0,0 +1,27 @@
+void ascon_encrypt(struct ascon_param *param);
+void ascon_decrypt(struct ascon_param *param);
+
+#endif
```
`#endif // define ASCON_H`

Denton-L committed:
libascon: remove busy-waiting

The hardware implementation will block until a result is returned. As a
result, remove busy-waiting as we no longer need to test if the result
is ready.
